weldx/core.py

```diff
@@ -1020,7 +1025,9 @@ def __init__(
         elif isinstance(obj, (MathematicalExpression, str, sympy.Expr)):
             if dims is not None and not isinstance(dims, dict):
                 raise ValueError(f"Argument 'dims' must be dict, not {dims}")
-            self._init_expression(obj, dims, parameters, units)
+            self._init_expression(
+                obj, dims, parameters, {k: U_(v) for k, v in units.items()}  # catch str
+            )
```

Comment:
I couldn't manage do simply do 
```python
units = {k: U_(v) for k, v in units.items()}
```
outside of the function call, but at least this does the trick @vhirtham